### ISSUE TITLE
Zone is not in Region using region-bound GCE machine auth

### ISSUE THREAD

**Issue by Luzifer** [hashicorp/vault-plugin-auth-gcp#36]
When using `bound_region` together with a `bound_instance_group` (at least) in region `europe-west3` the auth request against the plugin fails with an error a zone (clearly being in the region) is not in the region.

Might be related to #33?

Local Vault version: `v0.10.1`, Server: `v0.10.1` (from Hashicorp releases page)

### Configuration of the GCP role

```console
$ vault read auth/gcp/role/app-deploy
Key                       Value
---                       -----
bound_instance_group      app-region-group-manager
bound_labels              n/a
bound_region              europe-west3
bound_service_accounts    <nil>
bound_zone                n/a
max_ttl                   0
period                    0
policies                  [app-deploy]
project_id                ******
role_type                 gce
ttl                       0
```

### Sniplet from bash script performing the login

```bash
JWT=$(curl -s -G -H "Metadata-Flavor: Google" \
        --data-urlencode "audience=${VAULT_ADDR}/vault/${ROLE}" \
        --data-urlencode "format=full" \
        "http://metadata/computeMetadata/v1/instance/service-accounts/${SERVICE_ACCOUNT}/identity")

export VAULT_TOKEN=$(/tmp/vault write -field=token auth/gcp/login role=${ROLE} jwt=${JWT})
```

### Received error

```
Error writing data to auth/gcp/login: Error making API request.

URL: PUT https://******/v1/auth/gcp/login
Code: 400. Errors:

* zone 'europe-west3-b' is not in region 'https://www.googleapis.com/compute/v1/projects/******/regions/europe-west3'
```

**Comment by emilymye** [comment]
Fixed by #34

**Comment by petems** [comment]
Currently scheduled for release in Vault 0.10.2 👍
